src/platform/Module.c

```diff
@@ -0,0 +1,373 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <Module.h>
+#include <Log.h>
+
+#include <dlfcn.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include <parson.h>
+
+#define MMI_OPEN_FUNCTION "MmiOpen"
+#define MMI_CLOSE_FUNCTION "MmiClose"
+#define MMI_GET_FUNCTION "MmiGet"
+#define MMI_SET_FUNCTION "MmiSet"
+#define MMI_GETINFO_FUNCTION "MmiGetInfo"
+#define MMI_FREE_FUNCTION "MmiFree"
+
+// Required module info fields
+#define INFO_NAME "Name"
+#define INFO_DESCRIPTION "Description"
+#define INFO_MANUFACTURER "Manufacturer"
+#define INFO_VESRION_MAJOR "VersionMajor"
+#define INFO_VERSION_MINOR "VersionMinor"
+#define INFO_VERSION_INFO "VersionInfo"
+#define INFO_COMPONENTS "Components"
+#define INFO_LIFETIME "Lifetime"
+
+// Optional module info fields
+#define INFO_VERSION_PATCH "VersionPatch"
+#define INFO_VERSION_TWEAK "VersionTweak"
+#define INFO_LICENSE_URI "LicenseUri"
+#define INFO_PROJECT_URI "ProjectUri"
+#define INFO_USER_ACCOUNT_URI "UserAccount"
+
+void FreeModuleInfo(MODULE_INFO* info)
+{
+    if (NULL == info)
+    {
+        return;
+    }
+
```

Comment:
Rather, do the opposite, and if not null, then free. To avoid the early return that's not really necessary.